test/Microsoft.Extensions.Logging.Test/TypeNameHelperTest.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.Abstractions.Internal
+{
+    public class TypeNameHelperTest
+    {
+        [Fact]
+        public void Can_pretty_print_CLR_full_name()
+        {
+            // Predefined Types
+            Assert.Equal("int",
+                TypeNameHelper.GetTypeDisplayName(typeof(int)));
+            Assert.Equal("System.Collections.Generic.List",
+                TypeNameHelper.GetTypeDisplayName(typeof(List<int>)));
+            Assert.Equal("System.Collections.Generic.Dictionary",
+                TypeNameHelper.GetTypeDisplayName(typeof(Dictionary<int, string>)));
+            Assert.Equal("System.Collections.Generic.Dictionary",
+                TypeNameHelper.GetTypeDisplayName(typeof(Dictionary<int, List<string>>)));
+            Assert.Equal("System.Collections.Generic.List",
+                TypeNameHelper.GetTypeDisplayName(typeof(List<List<string>>)));
+
+            // Classes inside NonGeneric class
+            Assert.Equal("Microsoft.Extensions.Logging.Abstractions.Internal.TypeNameHelperTest.A",
```

Comment:
`A` here is a nested type actually...I suppose changing `+` to `.` is fine, even if the nested type is public or private?